### ISSUE TITLE
PDF Export

### ISSUE THREAD

**Issue by dignati** [getredash/redash#2861]
We would like the ability to export dashboards as PDFs both via UI and as an API call. 

I believe the only practical solution to generating PDFs would be to use a library like [pdfkit](https://pypi.org/project/pdfkit/) that basically screenshots websites. To generate PDFs directly we would need to replicate the javascript code that is rendering visualizations. 

Would you be open to a PR implementing this feature as described? Me and a coworker could start working on this soon.

**Comment by arikfr** [comment]
I think a better solution will be to use [puppeteer](https://github.com/GoogleChrome/puppeteer), which uses Chrome. Otherwise you will probably have rendering issues with wkhtmltopdf, which probably won't support some of the Javascript we use.

We're definitely open to such PR, but would like to discuss the implementation first to make sure it's something we can support in the long run.

Thanks!

**Comment by dignati** [comment]
Should we use [pyppeteer](https://github.com/miyakogi/pyppeteer) then?

**Comment by arikfr** [comment]
Nope, pyppeteer works with Python 3.6+, so won't work in the same process as Redash anyway.

**Comment by dignati** [comment]
So are you suggesting to run a nodejs process on the server to use puppeteer? I'm making process on adding basic UI to redash so that I can test a couple different backend approaches.

**Comment by arikfr** [comment]
On the server or (preferably) as an additional Docker container.

> I'm making process on adding basic UI to redash

basic ui for this feature?

**Comment by dignati** [comment]
Oh okay, additional container seems doable. 

> basic ui for this feature?

Yes. Basically just a download as pdf button in the dashboard context.
